subsys/lorawan/lorawan.c

```diff
@@ -359,6 +374,46 @@ int lorawan_set_region(enum lorawan_region region)
 	return 0;
 }
 
+int lorawan_request_time(bool force_request, bool sync_call)
+{
+	int ret = 0;
+	LoRaMacStatus_t status;
+
+	status = lorawan_req_time();
+	if (status != LORAMAC_STATUS_OK) {
+		LOG_ERR("DeviceTime Req. failed: %s",
+				lorawan_status2str(status));
+		ret = lorawan_status2errno(status);
+		goto out;
+	}
+
+	if (sync_call) {
+		k_sem_reset(&mlme_confirm_sem);
```

Comment:
Arbitrarily resetting this semaphore seems dangerous. What happens if this another action is already running from another thread?